geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseIndexAdapter.scala

```diff
@@ -301,110 +303,131 @@ class HBaseIndexAdapter(ds: HBaseDataStore) extends IndexAdapter[HBaseDataStore]
 
   /**
-    * Configure the hbase scan
-    *
-    * @param originalRanges ranges to scan
-    * @param colFamily col family to scan
-    * @param filters scan filters
-    * @param coprocessor coprocessor scan or not
-    * @return
-    */
-  protected def configureScans(originalRanges: Seq[Scan],
-                               colFamily: Array[Byte],
-                               filters: Seq[HFilter],
-                               coprocessor: Boolean): Seq[Scan] = {
-    import scala.collection.JavaConverters._
+   * Configure the hbase scan
+   *
+   * @param tables tables being scanned, used for region location information
+   * @param ranges ranges to scan, non-empty. needs to be mutable as we will sort it in place
+   * @param small whether 'small' ranges (i.e. gets)
+   * @param colFamily col family to scan
+   * @param filters scan filters
+   * @param coprocessor is this a coprocessor scan or not
+   * @return
+   */
+  protected def configureScans(
+      tables: Seq[TableName],
+      ranges: Seq[RowRange],
+      small: Boolean,
+      colFamily: Array[Byte],
+      filters: Seq[HFilter],
+      coprocessor: Boolean): Seq[Scan] = {
 
     val cacheBlocks = HBaseSystemProperties.ScannerBlockCaching.toBoolean.get // has a default value so .get is safe
     val cacheSize = HBaseSystemProperties.ScannerCaching.toInt
 
     logger.debug(s"HBase client scanner: block caching: $cacheBlocks, caching: $cacheSize")
 
-    if (!coprocessor && originalRanges.headOption.exists(_.isSmall)) {
+    if (small && !coprocessor) {
       val filter = filters match {
         case Nil    => None
         case Seq(f) => Some(f)
         case f      => Some(new FilterList(f: _*))
       }
-      originalRanges.map { s =>
-        val scan = new Scan(s).addFamily(colFamily)
+      ranges.map { r =>
+        val scan = new Scan(r.getStartRow, r.getStopRow)
+        scan.addFamily(colFamily).setCacheBlocks(cacheBlocks).setSmall(true)
         filter.foreach(scan.setFilter)
-        scan.setCacheBlocks(cacheBlocks)
         cacheSize.foreach(scan.setCaching)
         ds.applySecurity(scan)
         scan
       }
     } else {
-      val rowRanges = sortAndMerge(originalRanges)
+      // split and group ranges by region server
+      val rangesPerRegionServer = scala.collection.mutable.Map.empty[ServerName, ListBuffer[RowRange]]
 
-      // TODO GEOMESA-1806 parameterize this?
-      val rangesPerThread = math.min(ds.config.maxRangesPerExtendedScan,
-        math.max(1, math.ceil(rowRanges.size() / ds.config.queryThreads * 2).toInt))
-
-      // group scans into batches to achieve some client side parallelism
-      // we double the initial size to account for extra groupings based on the shard byte
-      val groupedScans = new java.util.ArrayList[Scan]((rowRanges.size() / rangesPerThread + 1) * 2)
+      WithClose(ds.connection.getRegionLocator(tables.head)) { locator =>
```

Comment:
Ok, if partitioning is used, what happens here?  It seems like you could end up scanning multiple tables, no?

Reply:
Shouldn't we have a per-table set of scans to run?  (Maybe we do?)